src/instana/propagators/base_propagator.py

```diff
@@ -232,15 +251,17 @@ def __determine_span_context(
             ctx_synthetic = synthetic
 
         if correlation:
-            ctx_correlation_type, ctx_correlation_id = self._get_correlation_properties(level)
+            ctx_correlation_type, ctx_correlation_id = self._get_correlation_properties(
+                level
+            )
 
         if traceparent:
             ctx_traceparent = traceparent
             ctx_tracestate = tracestate
 
         return SpanContext(
-            trace_id=ctx_trace_id,
-            span_id=ctx_span_id,
+            trace_id=int(ctx_trace_id),
+            span_id=int(ctx_span_id),
```

Comment:
Why is this change needed?

Reply:
Because span context is waiting for 2 int values for trace id and span id, and since we use get_instana_ancestor to provide these values, they may be strings (as you can see in the test case below)

<img width="1031" alt="image" src="https://github.com/user-attachments/assets/047ba653-a1dc-4fa1-8b3e-0f12ae7a8f61">

Reply:
Don't change that at this point. 
If you read the code, `ctx_trace_id` and `ctx_span_id` have been set to integers since the beginning (see lines 196 and 197).
If, as you mentioned, the values coming from the `get_instana_ancestor()` are string, then you have to change them by evaluating their type where `ctx_trace_id` and `ctx_span_id`  are overwritten on lines 227 and 228: 
```
if instana_ancestor:
    ctx_trace_id = int(instana_ancestor.t)
    ctx_span_id = int(instana_ancestor.p)
```

Reply:
get_traceparent_fields is also returning string below, we'll need to cast to integer it's values as well, you can see the tp_trace_id and tp_parent_id are being set to ctx_trace_id and ctx_span_id in disable_traceparent block:


 ```
  _, tp_trace_id, tp_parent_id, _ = self._tp.get_traceparent_fields(
        traceparent
    )

    if tracestate and "in=" in tracestate:
        instana_ancestor = self._ts.get_instana_ancestor(tracestate)

    if disable_traceparent == "":
        ctx_trace_id = tp_trace_id[-16:]
        ctx_span_id = tp_parent_id
        ctx_synthetic = synthetic
        ctx_trace_parent = True
        ctx_instana_ancestor = instana_ancestor
        ctx_long_trace_id = tp_trace_id
    else:
        if instana_ancestor:
            ctx_trace_id = int(instana_ancestor.t)
            ctx_span_id = int(instana_ancestor.p)
            ctx_synthetic = synthetic

```

Reply:
> get_traceparent_fields is also returning string below, we'll need to cast to integer it's values as well, you can see the tp_trace_id and tp_parent_id are being set to ctx_trace_id and ctx_span_id in disable_traceparent block:

So, go ahead and do it there as well! 😉

Reply:
Done 👍